serialization.md

```diff
@@ -385,21 +385,15 @@ Encoding
-                                             , VALUE
-                                             )
-                              )
-      => #rlpEncodeLength(         #rlpMerkleH( #rlpEncodeMerkleTree(  P0 ) ) +String #rlpMerkleH( #rlpEncodeMerkleTree(  P1 ) )
-                           +String #rlpMerkleH( #rlpEncodeMerkleTree(  P2 ) ) +String #rlpMerkleH( #rlpEncodeMerkleTree(  P3 ) )
-                           +String #rlpMerkleH( #rlpEncodeMerkleTree(  P4 ) ) +String #rlpMerkleH( #rlpEncodeMerkleTree(  P5 ) )
-                           +String #rlpMerkleH( #rlpEncodeMerkleTree(  P6 ) ) +String #rlpMerkleH( #rlpEncodeMerkleTree(  P7 ) )
-                           +String #rlpMerkleH( #rlpEncodeMerkleTree(  P8 ) ) +String #rlpMerkleH( #rlpEncodeMerkleTree(  P9 ) )
-                           +String #rlpMerkleH( #rlpEncodeMerkleTree( P10 ) ) +String #rlpMerkleH( #rlpEncodeMerkleTree( P11 ) )
-                           +String #rlpMerkleH( #rlpEncodeMerkleTree( P12 ) ) +String #rlpMerkleH( #rlpEncodeMerkleTree( P13 ) )
-                           +String #rlpMerkleH( #rlpEncodeMerkleTree( P14 ) ) +String #rlpMerkleH( #rlpEncodeMerkleTree( P15 ) )
+    rule #rlpEncodeMerkleTree ( MerkleBranch ( M , VALUE ) )
+      => #rlpEncodeLength(         #rlpMerkleH( #rlpEncodeMerkleTree( {M[0]  orDefault .MerkleTree}:>MerkleTree ) ) +String #rlpMerkleH( #rlpEncodeMerkleTree( {M[1]  orDefault .MerkleTree}:>MerkleTree ) )
+                           +String #rlpMerkleH( #rlpEncodeMerkleTree( {M[2]  orDefault .MerkleTree}:>MerkleTree ) ) +String #rlpMerkleH( #rlpEncodeMerkleTree( {M[3]  orDefault .MerkleTree}:>MerkleTree ) )
+                           +String #rlpMerkleH( #rlpEncodeMerkleTree( {M[4]  orDefault .MerkleTree}:>MerkleTree ) ) +String #rlpMerkleH( #rlpEncodeMerkleTree( {M[5]  orDefault .MerkleTree}:>MerkleTree ) )
+                           +String #rlpMerkleH( #rlpEncodeMerkleTree( {M[6]  orDefault .MerkleTree}:>MerkleTree ) ) +String #rlpMerkleH( #rlpEncodeMerkleTree( {M[7]  orDefault .MerkleTree}:>MerkleTree ) )
+                           +String #rlpMerkleH( #rlpEncodeMerkleTree( {M[8]  orDefault .MerkleTree}:>MerkleTree ) ) +String #rlpMerkleH( #rlpEncodeMerkleTree( {M[9]  orDefault .MerkleTree}:>MerkleTree ) )
+                           +String #rlpMerkleH( #rlpEncodeMerkleTree( {M[10] orDefault .MerkleTree}:>MerkleTree ) ) +String #rlpMerkleH( #rlpEncodeMerkleTree( {M[11] orDefault .MerkleTree}:>MerkleTree ) )
+                           +String #rlpMerkleH( #rlpEncodeMerkleTree( {M[12] orDefault .MerkleTree}:>MerkleTree ) ) +String #rlpMerkleH( #rlpEncodeMerkleTree( {M[13] orDefault .MerkleTree}:>MerkleTree ) )
+                           +String #rlpMerkleH( #rlpEncodeMerkleTree( {M[14] orDefault .MerkleTree}:>MerkleTree ) ) +String #rlpMerkleH( #rlpEncodeMerkleTree( {M[15] orDefault .MerkleTree}:>MerkleTree ) )
```

Comment:
The java backend is failing on `storageRoot-spec.k`. Could it have something to do with the use of `orDefault` here?